core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/RDFStarTripleSource.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.algebra.evaluation;
+
+import org.eclipse.rdf4j.common.iteration.CloseableIteration;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Triple;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.query.QueryEvaluationException;
+
+public interface RDFStarTripleSource extends TripleSource {
```

Comment:
This is minor, but since this is a behavioral interface it doesn't have to extend TripleSource - just thinking ahead of a possible future design where TripleSource and RDFStarTripleSource methods are not provided by the same object.

Reply:
The interface as is, does not provide any other meaningful uses for the classes that implement it. It makes sence only for those classes that also implement TripleSource that's why it extends TripleSource.

Reply:
I'm not sure I agree with that, but as said it's minor, fine to keep as-is, my "possible future design" may well be a YAGNI.